crates/ruma-common/src/events/content.rs

```diff
@@ -21,20 +21,16 @@ pub trait EventContent: Sized + Serialize {
 
     /// Get the event's type, like `m.room.message`.
     fn event_type(&self) -> Self::EventType;
-
-    /// Constructs the given event content.
-    #[doc(hidden)]
-    fn from_parts(event_type: &str, content: &RawJsonValue) -> serde_json::Result<Self>;
 }
 
 impl<T> Raw<T>
 where
-    T: EventContent,
+    T: EventContentFromType,
     T::EventType: fmt::Display,
 {
-    /// Try to deserialize the JSON as an event's content.
-    pub fn deserialize_content(&self, event_type: T::EventType) -> serde_json::Result<T> {
-        T::from_parts(&event_type.to_string(), self.json())
+    /// Try to deserialize the JSON as an event's content with the given event type.
+    pub fn deserialize_with_type(&self, event_type: T::EventType) -> serde_json::Result<T> {
+        <T as EventContentFromType>::from_parts(&event_type.to_string(), self.json())
```

Comment:
This qualification seems unnecessary to me after the last commit.

```suggestion
        T::from_parts(&event_type.to_string(), self.json())
```